stablehlo/dialect/TypeInference.cpp

```diff
@@ -948,6 +948,52 @@ LogicalResult inferSortOp(
   return success();
 }
 
+LogicalResult inferTransposeOp(MLIRContext* context, Optional<Location> loc,
+                               Value operand,
+                               llvm::SmallVector<int64_t>& operandBounds,
+                               DenseIntElementsAttr permutation,
+                               SmallVectorImpl<Type>& inferredReturnTypes,
+                               llvm::SmallVector<int64_t>& resultBounds) {
+  auto type = operand.getType();
+  auto rankedTy = type.dyn_cast<RankedTensorType>();
+  if (!rankedTy) {
+    auto shapedTy = type.dyn_cast<ShapedType>();
```

Comment:
is this cast necessary? inferredReturnTypes is of type `Type`?

Reply:
Fixed, thanks!